src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,144 @@
+        greetings();
+        readFile(lists);
+        processInput();
+        writeFile(lists);
+        goodbye();
+    }
+
+
+    public static void processInput(){
+        Scanner userInput = new Scanner(System.in);
+        while(true) {
+            inputString = userInput.nextLine().trim();
+            String[] inputStringSplit = inputString.split(" ", 2);
+            System.out.println(LINE);
+            if (inputString.equalsIgnoreCase("bye")) {
+                return;
+            } else if (inputStringSplit[0].equalsIgnoreCase("done")) {
+                if (inputStringSplit.length > 1) {
+                    MarkAsDone.markAsDone(inputStringSplit[1]);
+                } else {
+                    DukeException.doneWithoutNo();
+                }
+            } else if (inputStringSplit[0].equalsIgnoreCase("delete")) {
+                if (inputStringSplit.length > 1) {
+                    Delete.deleteTask(inputStringSplit[1]);
+                } else {
+                    DukeException.deleteWithoutNo();
+                }
+            } else if (inputStringSplit[0].equalsIgnoreCase("list")) {
+                printList(0, taskCount);
+            } else if (AddToList.keywordCheck(inputStringSplit[0])) {
+                if (inputStringSplit.length > 1) {
+                    AddToList.AddToList(inputStringSplit[0], inputStringSplit[1]);
+                } else {
+                    DukeException.taskDescriptionEmpty();
+                }
+            } else {
+                DukeException.illegalInput();
+            }
+        }
+    }
+
+    private static void readFile(List<Task> lists){
+        try {
+            File file = new File("src/main/java/Duke/Duke.txt");
+            if (file.createNewFile()) {
+                System.out.println("A new file has been created! ^_^");
+            } else {
+                System.out.println("Reading saved Task Lists ^_^");
+                Scanner readingFile = new Scanner(file);
+                while (readingFile.hasNextLine()) {
+                    String line = readingFile.nextLine();
+                    String[] parts = line.split("\\|", 4);
+                    String type = parts[0];
+                    String isDone = parts[1];
+                    String task = parts[2];
+                    if (type.equals("T")) {
+                        Task taskInFile = new ToDoTask(task);
+                        lists.add(taskInFile);
+                        if (isDone.equals("true")) {
+                            taskInFile.markAsDone();
+                        }
+                    } else if (type.equals("D")) {
+                        String time = parts[3];
+                        DeadlineTask taskInFile = new DeadlineTask(task, time);
+                        lists.add(taskInFile);
+                        if (isDone.equals("true")) {
+                            taskInFile.markAsDone();
+                        }
+                    } else {
```

Comment:
Try and explicitly state all what all your if and else-if conditions, and leave the 'else' statement for error handling mainly